signalfx-java/src/test/java/com/signalfx/metrics/connection/HttpDataPointProtobufReceiverConnectionTest.java

```diff
@@ -193,6 +252,20 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
     }
   }
 
+  private static class CountingTimeoutHandler extends AbstractHandler {
+    private int requests = 0;
+
+    @Override
+    public void handle(String s, Request request, HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws IOException{
+      requests++;
+
+      try {
+        Thread.sleep(2000);
```

Comment:
I'm always bummed to see tests getting slower...and in my experience tests with timeouts are often flappy/flaky when run in cloud CI/CD. Can we figure out a way to pass a `CountdownLatch` or something that the test/caller can trigger so that we don't have to actually wait?

Reply:
I know that feel 🫂 , I saw it being used already so chucked it in. I've modified it to a hybrid approach, a latched handler that also has a timeout. If the latch is triggered manually then it will exit before, for convenience, I'm providing a timeout value on the latch for my testing purposes.

Let me know if this is good enough, or if you'd prefer I trigger the latch manually.

Reply:
Na you're right, I opted to use `await` with the timeout instead, I've changed the test to schedule the release of the latch after a certain time period instead.

It's quite tricky with the retry test case, we have to release the latch after the retried request since they share the same latch. Currently I've scheduled the latch to `countDown()` after 3s. 3s was chosen because it's sufficient for the retry to trigger. Anything less and it might exit before the retry.

Reply:
Did we just turn a 2 second test into a 3 second test? 😁 
The coordination between threads for testing purposes can be a hassle, indeed. 

Maybe we can rethink this -- Seems like there are two testing goals here: 
1. to test the case where we *DO* retry when it's a retryable exception (not in the non-retryable list) 
2. to test the case were we *DO NOT* retry when the exception is in the non-retryable list.

For case 1, rather than depending on `SocketTimeoutException`, can we use a different exception? Something that fails almost immediately. I'm not sure offhand but there's gotta be something, right? Maybe remove `UnknownHostException` from the list for this test and force that?

For case 2, I think it should just be as simple as forcing that `UnknownHostException` with the default list, right?

Reply:
> For case 1, rather than depending on SocketTimeoutException, can we use a different exception? Something that fails almost immediately. I'm not sure offhand but there's gotta be something, right? Maybe remove UnknownHostException from the list for this test and force that?

While we can force a retry with UnknownHostException, it would be hard to assert that the retry has indeed taken place without some sort of mock server keeping track of request counts, or some sort of http client library to keep track of request counts (afaik we can't modify this for the DatapointPointReceiver).

Throwing an exception in the server/handler doesn't necessarily propagate back to the client either.

Let me see if there are any other retry scenarios that I can force without relying on socket timeouts.

Reply:
For now I've tuned the test case to trigger the latch after 500ms instead of 3s (by tuning the receiver timeout).

Reply:
Well, `HttpDataPointProtobufReceiverConnectionTest` failed in ci/cd. Womp womp.

Reply:
I think this will have to do for now. Thanks for sticking it out, I appreciate it.